src/main/kotlin/org/rust/openapiext/utils.kt

```diff
@@ -101,6 +102,15 @@ fun checkIsSmartMode(project: Project) {
     if (DumbService.getInstance(project).isDumb) throw IndexNotReadyException.create()
 }
 
+fun checkCommitIsNotInProgress(project: Project) {
+    val app = ApplicationManager.getApplication()
+    if ((app.isUnitTestMode || app.isInternal) && app.isDispatchThread) {
+        if ((PsiDocumentManager.getInstance(project) as PsiDocumentManagerBase).isCommitInProgress) {
+            error("Accessing indices during PSI event processing can lead to typing performance issues")
```

Comment:
Probably, `com.intellij.openapi.diagnostic.Logger#error` is better because it throws an error in tests and always write in log even in non-internal mode